### PR TITLE
add ignoreHttp1xx boolean field to health checks

### PR DESCRIPTION
**Pull request by nicgrayson**
Here is the list of fields on a health check. This should be one of them.
https://mesosphere.github.io/marathon/docs/health-checks.html

### CODE REVIEW

File: health.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ type HealthCheck struct {
 	GracePeriodSeconds     int      `json:"gracePeriodSeconds,omitempty"`
 	IntervalSeconds        int      `json:"intervalSeconds,omitempty"`
 	TimeoutSeconds         int      `json:"timeoutSeconds,omitempty"`
+	IgnoreHttp1xx          *bool    `json:"ignoreHttp1xx,ommitempty"`
```
Comment (timoreimann) [inline health.go]: Could you use all caps for HTTP in the variable name (but not the tag) please?

File: health.go [hunk 2]
```diff
@@ -59,11 +60,18 @@ func (h HealthCheck) SetMaxConsecutiveFailures(i int) HealthCheck {
 	return h
 }
 
+// SetIgnoreHttp1xx sets ignore http 1xx on the health check.
+func (h HealthCheck) SetIgnoreHttp1xx(b bool) HealthCheck {
```
Comment (timoreimann) [inline health.go]: Http -> HTTP here too please.

File: health.go [hunk 3]
```diff
@@ -59,11 +60,18 @@ func (h HealthCheck) SetMaxConsecutiveFailures(i int) HealthCheck {
 	return h
 }
 
+// SetIgnoreHttp1xx sets ignore http 1xx on the health check.
```
Comment (timoreimann) [inline health.go]: http -> HTTP.

File: health.go [hunk 4]
```diff
@@ -59,11 +60,18 @@ func (h HealthCheck) SetMaxConsecutiveFailures(i int) HealthCheck {
...
+	h.IgnoreHttp1xx = &b
+	return h
+}
+
 // NewDefaultHealthCheck creates a default application health check
 func NewDefaultHealthCheck() *HealthCheck {
 	portIndex := 0
 	path := ""
 	maxConsecutiveFailures := 3
+	ignoreHttp1xx := false
```
Comment (timoreimann) [inline health.go]: Do we need to set it here (and use it later)? Wouldn't we just continue to apply the default by omitting the value?

File: health.go (hunk 1 above)
Comment (nicgrayson) [inline health.go]: Yes

File: health.go (hunk 3 above)
Comment (nicgrayson) [inline health.go]: yes

File: health.go (hunk 4 above)
Comment (nicgrayson) [inline health.go]: Makes sense I'll fix

**Comment by nicgrayson** [comment]
Fixed naming issues, amended, and force pushed.

File: health.go [hunk 5]
```diff
@@ -59,6 +60,12 @@ func (h HealthCheck) SetMaxConsecutiveFailures(i int) HealthCheck {
 	return h
 }
 
+// SetIgnoreHTTP1xx sets ignore http 1xx on the health check.
+func (h HealthCheck) SetIgnoreHTTP1xx(b bool) HealthCheck {
+	h.IgnoreHttp1xx = &b
```
Comment (timoreimann) [inline health.go]: You forgot to rename this one too.

File: health.go [hunk 6]
```diff
@@ -59,6 +60,12 @@ func (h HealthCheck) SetMaxConsecutiveFailures(i int) HealthCheck {
 	return h
 }
 
+// SetIgnoreHTTP1xx sets ignore http 1xx on the health check.
+func (h HealthCheck) SetIgnoreHTTP1xx(b bool) HealthCheck {
```
Comment (timoreimann) [inline health.go]: While at it: can we rename `b` to something like `ignore` too?

Comment (nicgrayson) [inline health.go]: done

File: health.go (hunk 5 above)
Comment (nicgrayson) [inline health.go]: oops fixed

**Comment by timoreimann** [comment]
LGTM, thank you Nic!

**Pull request merged**